src/IROperator.cpp

```diff
@@ -185,6 +185,29 @@ bool is_negative_const(Expr e) {
     return false;
 }
 
+bool is_negatable_const(Expr e, Type T) {
+    if (T.is_uint()) return false;
+    if (const IntImm *i = e.as<IntImm>()) {
+        return !T.is_min(i->value);
+    }
+    if (e.as<FloatImm>()) return true;
+    if (const Cast *c = e.as<Cast>()) {
+        return is_negatable_const(c->value, c->type);
+    }
+    if (const Ramp *r = e.as<Ramp>()) {
+        // slightly conservative
+        return is_negatable_const(r->base) && is_negative_const(r->stride);
```

Comment:
Yeah, it's wrong. I'm tempted to just not support ramps here, because supporting it without being wildly over-conservative or using the simplifier (which seems like overkill here) is difficult. What do you think?